packages/lab/src/badge/Badge.tsx

```diff
@@ -60,32 +34,39 @@ export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
     window: targetWindow,
   });
 
-  const badgeId = useId();
-  const childId = useId(
-    isValidElement<HTMLAttributes<HTMLElement>>(children)
-      ? children.props?.id
-      : undefined
-  );
+  let valueText = value;
 
-  let badgeContentValue = badgeContent;
-  if (badgeContentValue > max) {
-    badgeContentValue = `${max}+`;
+  if (typeof value === "number" && max) {
+    valueText = value > max ? `${max}+` : value;
+  } else if (typeof value === "string") {
+    valueText = value.length > 4 ? `${value.slice(0, 4)}...` : value;
   }
 
-  return (
-    <span
-      className={clsx(withBaseName(), className)}
-      ref={ref}
-      role="img"
-      aria-labelledby={clsx(childId, badgeId)}
-      {...rest}
-    >
-      {children && isValidElement<HTMLAttributes<HTMLElement>>(children)
-        ? cloneElement(children, { id: childId })
-        : children}
-      <span id={badgeId} className={clsx(withBaseName("badge"))}>
-        {badgeContentValue}
+  if (!children) {
+    return (
+      <div>
+        <span
+          className={clsx(withBaseName(), withBaseName("inline"), className)}
+          ref={ref}
+          {...rest}
+        >
+          {valueText}
+        </span>
+      </div>
+    );
+  } else
+    return (
+      <span
+        className={clsx(withBaseName("wrap"), className)}
+        ref={ref}
+        {...rest}
+      >
+        {children}
+        <span
+          className={clsx(withBaseName(), withBaseName(`topRight`), className)}
```

Comment:
You should not apply className prop to both elements. Always to the top most one.